### PR TITLE
CWL workflow fastq2fasta/bam2fasta

### PR DESCRIPTION
**Pull request by inutano**
Implemented three workflows:

- fastq2fasta
  - one for those who already created bwa index, map fastq and run bam2fasta
- fastq2fasta-create-bwaindex
  - create bwa index from given reference fasta before mapping, map fastq and run bam2fasta
- bam2fasta
  - variant call by freebayes, use bcftools to polish vcf and convert to fasta

All tools and workflows are confirmed by `cwltool --validate` but not yet tested with real input data.

### CODE REVIEW

File: cwl/freebayes.cwl [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+cwlVersion: v1.1
+hints:
+  DockerRequirement:
+    dockerPull: "quay.io/biocontainers/freebayes:1.3.2--py37hc088bd4_0"
+baseCommand: freebayes
+arguments:
+  - --bam
+  - $(inputs.bam)
+  # - --region=$(inputs.contig):1-$(inputs.contig_end)
+  - --max-coverage
```
Comment (ekg) [inline cwl/freebayes.cwl]: Remove the --max-coverage argument

Comment (ekg) [reply cwl/freebayes.cwl]: Also we should set --ploidy 1

File: cwl/bcftools-consensus.cwl [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    dockerPull: "quay.io/biocontainers/bcftools:1.10.2--hd2cd319_0"
+requirements:
+  InitialWorkDirRequirement:
+    listing:
+      - $(inputs.vcf)
+      - $(inputs.vcf_index)
+baseCommand: bcftools
+arguments:
+  - consensus
+  - -i'QUAL>1 && (GT="AA" || GT="Aa")'
```
Comment (ekg) [inline cwl/bcftools-consensus.cwl]: If we set freebayes to haploid morning this won't make sense. I think it becomes 'QUAL > 1 && GT="A"'

**Review by ekg — COMMENTED** [review]
A few things need to change for the viral application.

File: cwl/freebayes.cwl (hunk 1 above)
Comment (inutano) [reply cwl/freebayes.cwl]: Done.

File: cwl/bcftools-consensus.cwl (hunk 2 above)
Comment (inutano) [reply cwl/bcftools-consensus.cwl]: Thanks! I've fixed the parameter.

**Comment by tetron** [comment]
I opened a new PR with more work on this https://github.com/hpobio-lab/viral-analysis/pull/11

**Comment by edawson** [comment]
Awesome!

**Pull request merged**